### PR TITLE
Fix/mismatch len lora

### PR DESCRIPTION
**Pull request by aolemila**
Pass:
- `python3 test/srt/lora/test_dynamic_lora.py` , annotated assertion in run_base_comparison_test.
- `python3 test/srt/lora/test_bgmv_backend.py`

## Checklist

- [ ] Please use English, otherwise it will be closed.
- [ ] The purpose of the PR, or link existing issues this PR will resolve.
- [ ] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/managers/scheduler.py [hunk 1]
```diff
@@ -1198,12 +1198,6 @@ def run_batch(self, batch: ScheduleBatch) -> GenerationBatchResult:
             )
 
             if self.enable_overlap:
-                # Pre-initialize ForwardBatch for overlap scheduling optimization
-                from sgl_jax.srt.model_executor.forward_batch_info import ForwardBatch
-
-                model_worker_batch.forward_batch = ForwardBatch.init_new(
-                    model_worker_batch, self.tp_worker.get_model_runner()
-                )
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/managers/scheduler.py:1206]: Why delete init forward_batch

Comment (aolemila) [reply python/sgl_jax/srt/managers/scheduler.py:1206]: Just change the position, not delete.

File: python/sgl_jax/srt/models/llama.py [hunk 2]
```diff
@@ -347,18 +348,19 @@ def __call__(
...
-                forward_batch,
-                token_to_kv_pool,
-                residual,
-            )
-            layers_kv_fused.append(kv_fused)
-            layers_callback_flag.extend(callback_flag)
+        with LoraBatchContext.set_batch(forward_batch):
+            for layer_id, layer in enumerate(self.layers):
+                if layer_id in self.layers_to_capture:
+                    aux_hidden_states.append(hidden_states + residual)
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/models/llama.py]: Could LoraContext placed in 
```
       def jitted_run_model(
            model_def,
            model_state_def,
            model_state_leaves,
            forward_batch,
            token_to_kv_pool,
            logits_metadata,
        ):
            model_state = jax.tree_util.tree_unflatten(model_state_def, model_state_leaves)
            model = nnx.merge(model_def, model_state)
            with LoraBatchContext.set_batch(forward_batch):
                 res= model(forward_batch, token_to_kv_pool, logits_metadata)
            return res
```

File: python/sgl_jax/srt/server_args.py [hunk 3]
```diff
@@ -1071,6 +1071,10 @@ def check_dynamic_lora_args():
                     if isinstance(self.lora_paths, dict):
                         # Dict format: {"name": "path", ...}
                         for name, path in self.lora_paths.items():
+                            if name == "0":
+                                raise ValueError(
+                                    f"0 exists in {self.lora_paths} as key, it is not allowed"
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/server_args.py]: Maybe add comment like: this key is a server-reserved symbol, used for requests that do not go through LoRA.

File: python/sgl_jax/srt/models/llama.py (hunk 2 above)
Comment (aolemila) [reply python/sgl_jax/srt/models/llama.py]: done

File: python/sgl_jax/srt/server_args.py (hunk 3 above)
Comment (aolemila) [reply python/sgl_jax/srt/server_args.py]: done
